alpakka/src/main/scala/com/hiya/alternator/BackoffStrategy.scala

```diff
@@ -0,0 +1,50 @@
+package com.hiya.alternator
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Random
+
+trait BackoffStrategy {
+  def getRetry(retry: Int): FiniteDuration
+}
+
+object BackoffStrategy {
+
+  // Not a really good random
+  private def random(upper: Long): Long = {
+    Math.abs(Random.nextLong()) % upper
+  }
+
+
+  private val retryMax = 30
+
+  final case class Static(delay: FiniteDuration) extends BackoffStrategy {
+    override def getRetry(retry: Int): FiniteDuration = delay
+  }
+
+  final case class FullJitter(cap: FiniteDuration, base: FiniteDuration) extends BackoffStrategy {
+    private val baseMs = base.toMillis
+    private val capMs = cap.toMillis
+
+    assert(baseMs >=0 && baseMs < Integer.MAX_VALUE)
```

Comment:
assert without error message, why it is assert and not required?

Reply:
Hmmm